wrappers/javascript/nodejs/src/library/bindings.ts

```diff
@@ -0,0 +1,185 @@
+import { refType } from 'ref-napi'
+
+import {
+  ByteBufferStruct,
+  FFI_ERRORCODE,
+  FFI_OBJECT_HANDLE,
+  StringListStruct,
+  CredRevInfoStruct,
+  FFI_OBJECT_HANDLE_PTR,
+  FFI_STRING,
+  FFI_INT8,
+  FFI_INT64,
+  CredentialEntryListStruct,
+  CredentialProveListStruct,
+  ObjectHandleListStruct,
+  FFI_STRING_PTR,
+  ByteBufferStructPtr,
+  I64ListStruct,
+  RevocationEntryListStruct,
+  FFI_INT8_PTR,
+  FFI_VOID,
+} from '../ffi'
+
+export const nativeBindings = {
+  // first element is method return type, second element is list of method argument types
+  anoncreds_buffer_free: [FFI_VOID, [ByteBufferStruct]],
+  anoncreds_create_credential: [
+    FFI_ERRORCODE,
+    [
+      FFI_OBJECT_HANDLE,
+      FFI_OBJECT_HANDLE,
+      FFI_OBJECT_HANDLE,
+      FFI_OBJECT_HANDLE,
+      StringListStruct,
+      StringListStruct,
+      StringListStruct,
+      FFI_STRING,
+      refType(CredRevInfoStruct),
+      FFI_OBJECT_HANDLE_PTR,
+      FFI_OBJECT_HANDLE_PTR,
+      FFI_OBJECT_HANDLE_PTR,
+    ],
+  ],
+  anoncreds_create_credential_definition: [
+    FFI_ERRORCODE,
+    [
+      FFI_STRING,
+      FFI_OBJECT_HANDLE,
+      FFI_STRING,
+      FFI_STRING,
+      FFI_STRING,
+      FFI_INT8,
+      FFI_OBJECT_HANDLE_PTR,
+      FFI_OBJECT_HANDLE_PTR,
+      FFI_OBJECT_HANDLE_PTR,
+    ],
+  ],
+  anoncreds_create_credential_offer: [
+    FFI_ERRORCODE,
+    [FFI_STRING, FFI_STRING, FFI_OBJECT_HANDLE, FFI_OBJECT_HANDLE_PTR],
+  ],
+  anoncreds_create_credential_request: [
+    FFI_ERRORCODE,
+    [
+      FFI_STRING,
+      FFI_OBJECT_HANDLE,
+      FFI_OBJECT_HANDLE,
+      FFI_STRING,
+      FFI_OBJECT_HANDLE,
+      FFI_OBJECT_HANDLE_PTR,
+      FFI_OBJECT_HANDLE_PTR,
+    ],
+  ],
```

Comment:
Maybe a lot of work, but if we were to assign custom types based on the ffi types , it would be much easier which parameters should be passed.

E.g. let's say we have a method that takes a credentialDefinitionId. Currently it would be typed as:

```
  anoncreds_create_credential_request: [
    FFI_ERRORCODE,
    [
      FFI_STRING,
    ],
  ],
```

However we could also write it like this:

```
const FFI_CREDENTIAL_DEFINITION_ID = FFI_STRING

  anoncreds_create_credential_request: [
    FFI_ERRORCODE,
    [
      FFI_CREDENTIAL_DEFINITION_ID,
    ],
  ],
```

Just a thought, maybe impractical

Reply:
Yeah I think I had that for a bit, but keeping that across all the shared components was a huuuuge amount of effort. It makes sense a lot for readability, but it will only be needed in the `NodeJsAnoncreds` package, so it was not worth the effort me.